src/ray/core_worker/context.cc

```diff
@@ -86,6 +90,10 @@ struct WorkerThreadContext {
   /// The task ID for current task.
   TaskID current_task_id_;
 
+  /// Unique bytes to use for any ObjectIDs generated by the
+  /// current task (via ray.put or .remote calls).
+  TaskID current_task_unique_bytes_;
```

Comment:
nit: It's a bit weird that it's called unique_bytes but the type is TaskID. What about current_task_internal_id?